src/components/SkillTopics.tsx

```diff
@@ -0,0 +1,87 @@
+import { useMutation, useQuery } from "@apollo/client";
+import { useContext } from "react";
+import { GetTopicsInfosQuery } from "../generated/graphql";
+import {
+  ADD_SKILL_TO_TOPIC,
+  DELETE_SKILL_TO_TOPIC,
+} from "../graphql/mutations/skills";
+import { GET_TOPICS_INFOS } from "../graphql/queries/topics";
+import { useDarkMode } from "../utils/darkMode";
+import { i18nContext } from "../utils/i18nContext";
+import { FetchedSkill } from "../utils/types";
+
+type skillTopicsProps = {
+  refetch: any;
+  readOnly: boolean;
+  skill: FetchedSkill;
+  skillTopics: any;
+};
+
+const SkillTopics = ({
+  refetch,
+  readOnly,
+  skill,
+  skillTopics,
+}: skillTopicsProps) => {
+  const { t } = useContext(i18nContext);
+  const { darkMode } = useDarkMode();
+  const { data: topics, loading: loadingTopics } =
+    useQuery<GetTopicsInfosQuery>(GET_TOPICS_INFOS);
+
+  const [insertTopic] = useMutation(ADD_SKILL_TO_TOPIC);
+  const [deleteTopic] = useMutation(DELETE_SKILL_TO_TOPIC);
+
+  const updateTopic = (selectedTopic: { id: string; name: string }) => {
+    const skilltopic = skillTopics.SkillTopic?.find(
+      (value) => value.topicId === selectedTopic?.id
+    );
+    if (!skilltopic) {
```

Comment:
For me, this is useless, as we already have a valid id. IDs received are predefined by the previous select query and then no user free input

Reply:
I use this .find() two times : one for change css and another one to select the right request -> delete or add . For me this find is not useless but the process can be improve

Reply:
Ok I got it. Maybe you can use the .find() only one time in the map() function (line 69), then use the result for css and to pass an updateTopic or deleteTopic method in the onClick prop ?

Reply:
hmm if we use this in the map, we would select only the prefered topics but we want all the topics for the render. I'm sorry I didn't find out an easy way to resolve it 💩 if you have the solution we can implement it, or we can add a commentary in the [issue](https://github.com/Zenika/skillZ/issues/315) that will refacto this component

Reply:
I'll see what I can do in https://github.com/Zenika/skillZ/issues/315 otherwise I'll leave it like that 👍🏻